### PR TITLE
adds longestrunsubsequence

### PR DESCRIPTION
**Pull request by maxried**
This is an implementation of the longest run subsequence problem, introduced in (Schrinner et al., WABI 2020) https://drops.dagstuhl.de/opus/volltexte/2020/12795/. It describes a string problem, which looks for the longest subsequence of a string such that this sequence contains at most one consecutive run for each character in the underlying alphabet. The code contains two different algorithms (based on Integer Linear Programming and on Dynamic Programming) as described in the publication. The problem appears in the context of homology-based scaffolding of two contig sets, originating from two individuals of the same species.

### CODE REVIEW

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

File: recipes/longestrunsubsequence/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  noarch: python
+  script: "{{ PYTHON }} -m pip install --no-deps -vv ."
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
+  run:
+    - python >=3.5
+  run_constrained:
```
Comment (dpryan79) [inline recipes/longestrunsubsequence/meta.yaml:22]: Please remove the `run_constrained:` entry, it isn't needed. Having `pulp >=2` is sufficient.

**Review by dpryan79 — CHANGES_REQUESTED** [review]
See comment, just one small change and this is good to go.

File: recipes/longestrunsubsequence/meta.yaml (hunk 1 above)
Comment (maxried) [reply recipes/longestrunsubsequence/meta.yaml:22]: Thanks! My understanding of run_constrained is that it's meant to specify optional dependencies. longestrunsubsequence uses pulp if it detects its presence, but has no hard dependency on it. What's the proper way to specify this?

Comment (dpryan79) [reply recipes/longestrunsubsequence/meta.yaml:22]: OK, that should presumably work then. We usually just add optional requirements that we expect users will want as normal run-time dependencies.

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**